Makefile

```diff
@@ -0,0 +1,41 @@
+.PHONY: clean-pyc clean-build help test
+.DEFAULT_GOAL := help
+
+help: ## print this help screen
+	@perl -nle'print $& if m{^[a-zA-Z0-9_-]+:.*?## .*$$}' $(MAKEFILE_LIST) | sort | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-25s\033[0m %s\n", $$1, $$2}'
+
+clean: clean-build clean-pyc
+	@echo "all clean now .."
+
+clean-build: ## remove build artifacts
+	@rm -fr build/
+	@rm -fr dist/
+	@rm -fr htmlcov/
+	@rm -fr *.egg-info
+	@rm -rf .coverage
+
+clean-pyc: ## remove Python file artifacts
+	@find . -name '*.pyc' -exec rm -f {} +
+	@find . -name '*.pyo' -exec rm -f {} +
+	@find . -name '*.orig' -exec rm -f {} +
+	@find . -name '*~' -exec rm -f {} +
+
+release: clean ## package and upload a release (working dir must be clean)
+	@while true; do \
+		CURRENT=`python -c "import wagtail_modeltranslation; print(wagtail_modeltranslation.__version__)"`; \
+		echo ""; \
+		echo "=== The current version is $$CURRENT - what's the next one?"; \
+		echo "==========================================================="; \
+		echo "1 - new major version"; \
+		echo "2 - new minor version"; \
+		echo "3 - patch"; \
+		echo ""; \
+		read yn; \
+		case $$yn in \
+			1 ) bumpversion major; break;; \
+			2 ) bumpversion minor; break;; \
+			3 ) bumpversion patch; break;; \
+			* ) echo "Please answer 1-3.";; \
```

Comment:
If I'm reading this correctly one always has to bump the version upon release. What if the version was previously bumped in a commit and one wants to publish the current version?

Reply:
well, id say: test this workflow because you should only tag/bump the version once travis reports the last commit to be good and not before that point so bumping upfront make no sense. And after that its one command line call to do it all. Its so easy - you dont want to do it differently. :)

But even if after trying it out you still want to have the version bump separated, tell me and i will split the command!

Reply:
Tagging should only be done at the last moment, agreed. Some projects bump the version after releasing, so the version in git is always the next one. I believe they do this so if you install from git using `pip install -e ssh...` it's clear that the installed version is not the one from pypi (but that's my own theory). Also how would you go about releasing beta or alpha versions with this workflow?

I can't really tell you to do one way or the other, I'm mostly interested in the pros and cons of different approaches. Releases are handled by infoportugal (@DiogoMarques29) so I'd wait for their feedback before changing anything. 😉

Reply:
Tagging the repo to be one version ahead? I have not seen this so far but anyway id prefer having a newer current version than being held back on the real update for the next version that pip thinks i already have .. i personally would not do that and it sounds weird!

You have a point tho on the alpha and beta versions - my lil script cant handle that in the current state. The bumpversion package im using can handle this, so its upgradable, but then again this package is marked 0.8.1 which is beta all the way .. having an alpha beta or beta beta is even more weird. Its funny to look at this that way - but thinking about alpha and beta didnt even cross my mind before your question. 

I just wanted to easy the process of releasing, since i pushed another fix id like to have published and i know the process always was a hassle for me before i made that very addition to my makefiles. Its not perfect, but way more convenient than putting the version tag in, making a new commit, tagging it, compiling the new package and upload it to pypi.

I promise to integrate prerelease options once this package hits 1.0.0!
Just somebody answer to this PR to make it happen :)

Reply:
Hi @kakulukia,

An example of a repo that has its version ahead of the release is Wagtail. [Current version](https://github.com/wagtail/wagtail/blob/master/wagtail/__init__.py#L5) in repo reads `2.1.0-alpha.0` but no equivalent tag or release for it exists.

My suggestion is to split the release command into 3 (names are indicative only):

- `bumpversion`: bumps the version
- `release`: releases current version (without touching version)
- `bumpversionAndRelease`: runs both commands.

This way you have the advantage of the single command (as you intend) without sacrificing control or flexibility.

Reply:
okay, will change that .. im not happy with the overly long new name tho
What about:
- release: bumpversion + upload
- upload: building and uploading to pypi

Reply:
I'm not a fan of `bumpversionAndRelease` either. 🙂  But I find `release` and `upload` confusing since both do a similar thing: "publish" to pypi. Sharing the same name plus a suffix or prefix would make it clearer to the developer what's happening. Perhaps `release` and `releaseNewVersion`?

@DiogoMarques29, care to pitch in? You'll probably be the main user of the new `make` commands @kakulukia is adding so your opinion is important.

Reply:
That actually is an awesome idea!
Should be possible somehow!

Andy BTW: just calling make will give you a help screen like that

![bildschirmfoto 2018-04-16 um 12 39 32](https://user-images.githubusercontent.com/96352/38804686-8c377f82-4173-11e8-8389-5eb19bf5f8a8.png)

Reply:
Hello guys! I belive @dmarcelino suggestion is a good way to approach the problem. I also agree the names are a bit confusing. What about a interactive way of doing it, Is that a hard thing to do? I personally don't have any experience with makefiles!